reconcile/utils/imap_client.py

```diff
@@ -0,0 +1,54 @@
+import imaplib
+from typing import Union
+
+from reconcile.utils.secret_reader import SecretReader
+
+
+class ImapClient:
+    def __init__(self, settings):
```

Comment:
can you add a typehint `dict[str,Any]`. this will help spot this line again once we decide to finally replace the return value of `queries.get_app_interface_settings()` with a dataclass

please look out for other places where typehints and return typehints are missing